### PR TITLE
Parameter to override a field type on a per-language basis

### LINKED ISSUE #22: Macro params to override generated type
Thanks for the tool!
Currently this struct
```rust
#[typeshare]
pub struct OrderItem {
  #[ts(type = "any | null")]
  pub meta: Option<Value>,
}
```
will generate this ts type
```typescript
export interface OrderItem {
	meta?: Value;
}
```
where `Value` is `serde_json::Value`
I wonder if it's a good idea to support something like this
```rust
#[typeshare]
pub struct OrderItem {
  #[typeshare(type = "any | null")]
  pub meta: Option<Value>,
}
```

### PR DESCRIPTION
**Pull request by InquisitivePenguin**
Resolves #22.

There is now a field decorator that can override the type of the field in the generated language(s) of your choice. This override takes precedent over everything else, including type mapping.

### CODE REVIEW

**Comment by InquisitivePenguin** [comment]
This PR depends on #118.

File: core/data/tests/can_override_types/input.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#[typeshare]
+#[serde(rename_all = "camelCase")]
+struct OverrideStruct {
+    // These annotations are intentionally inconsistent across languages
+    #[typeshare(swift(type = "Int"), typescript(readonly, type = "any | undefined"), kotlin(type = "Int"), go(type = "uint"), scala(type = "Short"))]
```
Comment (Lucretiel) [inline core/data/tests/can_override_types/input.rs:5]: Nit: recommend splitting these up over multiple lines

**Comment by Lucretiel** [comment]
General questions:

- Is this the first `typeshare` attribute (aside from the top-level one)?
- Is there an assumption here that the types should be compatible without any additional (de)serialize logic, in either direction, in either rust or the target language?

**Review by Lucretiel — CHANGES_REQUESTED** [review]

**Comment by InquisitivePenguin** [comment]
> General questions:
> 
>     * Is this the first `typeshare` attribute (aside from the top-level one)?
> 
>     * Is there an assumption here that the types should be compatible without any additional (de)serialize logic, in either direction, in either rust or the target language?

* No, we already have field decorators like `#[typeshare(typescript(readonly))]`.
* Correct, we don't check for compatibility. That's the responsibility of the end-user (given that this is _usually_ a last-resort option)

**Review by Lucretiel — APPROVED** [review]

**Comment by Lucretiel** [comment]
Slightly screwed up merging #118 so I force-rebased this branch. Merge should be fine.

**Pull request merged**